crates/engine/tree/src/tree/mod.rs

```diff
@@ -62,6 +63,16 @@ mod metrics;
 use crate::{engine::EngineApiRequest, tree::metrics::EngineApiMetrics};
 pub use config::TreeConfig;
 
+/// This exists only to share the hook definition across many function definitions easily
+pub type InvalidBlockHook = Box<
+    dyn Fn(
+            SealedBlockWithSenders,
+            SealedHeader,
+            BlockExecutionOutput<Receipt>,
+            Option<(TrieUpdates, B256)>,
+        ) + Send,
+>;
```

Comment:
not a big fan of type alias for this, which makes for awkward `Box::new(|_, _, _, _| {})` calls.

could we convert this into a trait with a noop impl

Reply:
done, lmk if I got the idea right